test/app.test.js

```diff
@@ -438,12 +438,20 @@ describe('app', function() {
         params: null
       });
 
+      app.middlewareFromConfig(handlerFactory, {
+        enabled: true,
+        phase: 'routes:before',
+        methods: ['get', 'head'],
+        params: {x: 1}
+      });
```

Comment:
I believe this test will pass even without the implementation above. To verify that the method/verb filter works as intended, you need to test the negative case too (i.e. send a request with a method that is not allowed).